server/migrations/20180806205808-create-rating.js

```diff
@@ -8,21 +8,28 @@ module.exports = {
     },
     userId: {
       type: Sequelize.INTEGER,
-      allowNull: false
+      allowNull: false,
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+      references: {
+        model: 'Users',
+        key: 'id',
+        as: 'userId'
+      },
     },
-    articleId: {
-      type: Sequelize.INTEGER,
+    articleSlug: {
```

Comment:
@Veraclins well done on this! but do you mind to check? I think using the primary key over the slug for the association would be better. I think so because although we are 96% certain that the slug is unique but there we ascertained during one of our TTL interaction that there can be an occasion where the random string returned in the slug might not be unique. Using the articleId is best because it is primary, I think is better and more accurate.

Reply:
Thanks for your comment, but this is not actually true. First of all, the slug is unique in the database which means that it is not possible to save two articles with the same slug in the first place. Secondly, the slug is the only thing identifying the article on the frontend and as such we only get access to the after getting the article (which must be with the slug) from the database. I don't think it makes sense to make an extra db call using the slug which is the unique identifier of the article only because we want to attach the ID to the rating just to ensure uniqueness when we can already ensure uniqueness with the slug without any extra db call. That is like doubling the request time for no reason.